lib/maplibre.ex

```diff
@@ -213,6 +262,26 @@ defmodule MapLibre do
     end
   end
 
+  defp validate_data!(data, coordinates) do
+    if Table.Reader.impl_for(data),
+      do: validate_coordinates(coordinates),
+      else: raise(ArgumentError, "unsupported data")
```

Comment:
```suggestion
    _ = Table.Reader.impl_for!(data)
    validate_coordinates(coordinates)
```

But I think you can skip the `impl_for!` altogether and just let it raise later when you invoke the protocol.